salt/grains/core.py

```diff
@@ -983,6 +984,59 @@ def _virtual(osdata):
         )
     return grains
 
+def _virtual_hv(osdata):
+    '''
+    Returns detailed hypervisor information from sysfs
+    Currently this seems to be used only by Xen
+    '''
+    grains = {}
+
+    # Bail early if we're not running on Xen
+    if 'xen' not in osdata['virtual']:
+        return grains
+
+    # Try to get the exact hypervisor version from sysfs
+    try:
+        version = {}
+        for file in ('major', 'minor', 'extra'):
+            with salt.utils.fopen('/sys/hypervisor/version/{}'.format(file), 'r') as fhr:
```

Comment:
Can you use the `salt.utils.files.fopen` path here, instead of the `salt.utils.fopen`? We've split out the utils init file into separate files in 2018.3 and develop.

Reply:
Right. In that case you might also want to update your linter messages as that referes to salt.utils.fopen()